File: src/typescript/logic/compute/calculateTotalCompensation.ts
interface Employee {
    baseSalary: number;
    bonusPercentage: number;
    taxPercentage: number;
}

function calculateTotalCompensation(employee: Employee): number {
    const bonus = employee.baseSalary * employee.bonusPercentage;
    const tax = bonus * employee.taxPercentage;
    return employee.baseSalary + bonus - tax;
}

// Usage
const employee: Employee = {
    baseSalary: 50000,
    bonusPercentage: 0.10, // 10%
    taxPercentage: 0.20   // 20%
};

console.log(calculateTotalCompensation(employee));
\ No newline at end of file

File: truth_set/go/logic/compute/calculateTotalCompensation.go
package compute

type Employee struct {
    BaseSalary      float64
    BonusPercentage float64
    TaxPercentage   float64
}

func calculateTotalCompensation(employee Employee) float64 {
    bonus := employee.BaseSalary * employee.BonusPercentage
    tax := bonus * employee.TaxPercentage
    return employee.BaseSalary + bonus - tax
}